compiler/rustc_trait_selection/src/solve/trait_goals.rs

```diff
@@ -312,9 +313,11 @@ impl<'tcx> assembly::GoalKind<'tcx> for TraitPredicate<'tcx> {
         Self::consider_implied_clause(
             ecx,
             goal,
-            ty::Binder::dummy(
-                tcx.mk_trait_ref(goal.predicate.def_id(), [self_ty, generator.resume_ty()]),
-            )
+            ty::Binder::dummy(ty::TraitRef::new(
```

Comment:
`ty::TraitRef::new` starts getting pretty verbose, maybe we should start with some `use ty::TraitRef;`? unsure about the types opinion on that